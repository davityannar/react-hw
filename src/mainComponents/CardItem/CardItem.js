import React from 'react'
import { Link } from 'react-router-dom'
import './CardItem.scss'


const CardItem = (props) => {
  return (
    <div className='card'>
      <div className='card-image'>        
        <Link to={`/movies/${props.movieId}`} ><img src={props.imgSrc} /></Link>
      </div>
      <div className='card-description'>
        <div className='card-film-name' >
          <p className='card-film-title'><Link to={`/movies/${props.movieId}`}>{props.movieName}</Link></p>
          <span className='card-vote'>{props.movieVote}</span>
          <span className='card-year'>{props.movieAge}</span>
        </div>
        <p className='card-genre'>{props.movieGenre}</p>
      </div>
    </div>
  )
}

export default CardItem;





 
