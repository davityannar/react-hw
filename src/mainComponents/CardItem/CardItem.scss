.card {
  width: 30%;
  margin-bottom: 30px;
  margin-right: 5%;
  &:nth-child(3n) {
    margin-right: 0;
  }
  &:hover {
    cursor: pointer;
  }
}
.card-image {
  img {
    width: 100%;
    height: auto;
    margin-bottom: 10px;
  }
}
.card-film-name {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 5px;
  text-transform: uppercase;
  font-size: 20px;

  .card-film-title {
    flex-grow: 1;
  }

  .card-year,
  .card-vote {
    margin-left: 10px;
    padding: 3px 5px;
    border: 1px solid #000;
    border-radius: 5px;
  }
}
.card-genre {
  color: gray;
}
.card-description {
  width: 90%;
  margin: 0 auto;
}