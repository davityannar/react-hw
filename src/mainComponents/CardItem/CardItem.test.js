import React from 'react';
import CardItem from './CardItem';

describe('Test CardItem component', () => {
  it('Matching snapshot', () => {

    const props = {
        title: 'title',
        poster_path: 'http://test.am',
        release_date: '2018-03-30',
        genres: ['example']
    };

    const component = shallow(<CardItem { ...props } />);

    expect(component).toMatchSnapshot();
  })
})