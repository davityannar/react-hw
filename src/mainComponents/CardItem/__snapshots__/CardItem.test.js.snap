// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CardItem component Matching snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <img
      src="http://test.am"
    />
  </div>
  <div
    className="card-description"
  >
    <p
      className="card-film-name"
    >
      title
      <span
        className="card-year"
      >
        2018
      </span>
    </p>
    <p
      className="card-genre"
    >
      example
    </p>
  </div>
</div>
`;
